lib/combine_pdf/pdf_public.rb

```diff
@@ -473,6 +473,39 @@ def stamp_pages(stamp, options = {})
       end
     end
 
+    # Overwrites any existing outline given a title, page_number mapping
+    def create_page_outline(outline_entries, document_title = "Document")
```

Comment:
I think it might be good here to break some of this into other methods. What do you think about something like this?

```ruby
def create_page_outline(outline_entries, document_title = "Document")
  return if outline_entries.empty?

  doc_outline = { Title: document_title }
  root_outline = {
    Type: :Outlines,
    First: { referenced_object: doc_outline, is_reference_only: true },
    Last: { referenced_object: doc_outline, is_reference_only: true },
    Count: 1
  }

  page_entries = build_page_entries(outline_entries, doc_outline)
  add_page_links(page_entries, doc_outline)

  @outlines = root_outline
end

def build_page_entries(outline_entries, doc_outline)
  page_entries = []

  outline_entries.each do |outline|
    page_entries << {
      Title: outline[:title],
      Parent: { referenced_object: doc_outline, is_reference_only: true },
      Dest: { referenced_object: pages[outline[:page_number]], is_reference_only: true },
    }
  end

  page_entries
end

def add_page_links(page_entries, doc_outline)
  page_entries.each_with_index do |page, idx|
    unless idx == 0
      p[:Prev] = { referenced_object: page_entries[idx - 1], is_reference_only: true }
    end

    unless idx == page_entries.length - 1
      p[:Next] = { referenced_object: page_entries[idx + 1], is_reference_only: true }
    end
  end

  doc_outline[:First] = { referenced_object: page_entries.first, is_reference_only: true }
  doc_outline[:Last] = { referenced_object: page_entries.last, is_reference_only: true }
end
```

Reply:
I think that makes sense, definitely makes it a little clearer what each functional part is doing.